### PR TITLE
Ksagiyam/io do not use pickle

### PR DESCRIPTION
**Pull request by ksagiyam**
Use `repr` instead of pickling.

Involves a minor fix in UFL, so depends on https://github.com/firedrakeproject/ufl/pull/34

### CODE REVIEW

File: firedrake/bcs.py [hunk 1]
```diff
@@ -347,11 +347,11 @@ def function_arg(self, g):
             try:
                 g = as_ufl(g)
                 self._function_arg = g
-            except UFLException:
+            except BaseException:
```
Comment (dham) [inline firedrake/bcs.py:350]: `ValueError`

**Review by dham — CHANGES_REQUESTED** [review]
minor quibble.

File: firedrake/bcs.py (hunk 1 above)
Comment (ksagiyam) [reply firedrake/bcs.py]: Fixed.

**Comment by ksagiyam** [comment]
Merged as discussed.

**Pull request merged**